### PR TITLE
Refactor Digital Ocean Domain

### PR DESCRIPTION
**Pull request by Akasurde**
##### SUMMARY
* Remove dopy dependency
* Use DigitalOceanHelper class
* Incorporated BondAnthony's changes

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
lib/ansible/modules/cloud/digital_ocean/digital_ocean_domain.py

##### ANSIBLE VERSION
```
2.5devel
```

### CODE REVIEW

**Comment by Akasurde** [comment]
cc @BondAnthony

**Comment by Akasurde** [comment]
ready_for_review

**Comment by BondAnthony** [comment]
@Akasurde I'm currently testing this and working through a couple bugs. Quick glance is showing a failure on reading the response for adding a domain, line 115.

**Comment by Akasurde** [comment]
@BondAnthony I have incorporated your suggestions. Please review again.

**Comment by BondAnthony** [comment]
@Akasurde please see PR https://github.com/Akasurde/ansible/pull/4 for additional updates to this module.

**Comment by Akasurde** [comment]
@BondAnthony I have updated PR, please review and let me know.

**Comment by Akasurde** [comment]
@BondAnthony Thanks for review and contribution in this PR.

File: lib/ansible/modules/cloud/digital_ocean/digital_ocean_domain.py [hunk 1]
```diff
@@ -79,151 +79,119 @@
...
+        module.exit_json(changed=delete_event)
 
 
 def main():
     module = AnsibleModule(
         argument_spec=dict(
             state=dict(choices=['present', 'absent'], default='present'),
-            api_token=dict(aliases=['API_TOKEN'], no_log=True),
+            oauth_token=dict(
+                aliases=['API_TOKEN'],
```
Comment (adamwight) [inline lib/ansible/modules/cloud/digital_ocean/digital_ocean_domain.py:191]: This broke backwards compatibility for me.  I think it should be lowercase `'api_token'`?

**Pull request merged**